apis/v1beta1/gateway_types.go

```diff
@@ -159,9 +184,6 @@ type GatewaySpec struct {
 	// other networking infrastructure, or some other address that traffic will
 	// be sent to.
 	//
-	// The .listener.hostname field is used to route traffic that has already
-	// arrived at the Gateway to the correct in-cluster destination.
-	//
```

Comment:
I removed this because it's just confusing to talk about `hostname` here - it's not relevant at all.